tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+            for (var i = 0; i < nestedTypes.Length; i++)
+            {
+                var typeHandle = nestedTypes[i];
+                if (typeHandle.IsNil)
+                {
+                    continue;
+                }
+
+                if (DatadogMetadataReader.IsCompilerGeneratedAttributeDefinedOnType(typeHandle.RowId))
+                {
+                    continue;
+                }
+
+                if (!TryGetClassSymbols(typeHandle, out var nestedClassScope))
+                {
+                    continue;
+                }
+
+                nestedClassScopes[index] = nestedClassScope.Value;
+                index++;
+            }
+        }
+
+        private LanguageSpecifics GetClassLanguageSpecifics(TypeDefinition type)
+        {
+            var interfaceNames = GetClassInterfaceNames(type);
+
+            var baseClassNames = GetClassBaseClassNames(type);
+
+            var accessModifiers = (type.Attributes & TypeAttributes.VisibilityMask);
+            var classLanguageSpecifics = new LanguageSpecifics
+            {
+                AccessModifiers = accessModifiers > 0 ? new[] { accessModifiers.ToString() } : null,
+                Interfaces = interfaceNames,
+                SuperClasses = baseClassNames
+            };
+            return classLanguageSpecifics;
+        }
+
+        private Symbol[] GetFieldSymbols(FieldDefinitionHandleCollection fieldsCollection, TypeDefinition parentType)
+        {
+            int index = 0;
+            Model.Symbol[] fieldSymbols = new Model.Symbol[fieldsCollection.Count];
+            foreach (var fieldDefHandle in fieldsCollection)
+            {
+                var fieldDef = MetadataReader.GetFieldDefinition(fieldDefHandle);
+                var fieldTypeName = fieldDef.DecodeSignature(new TypeProvider(false), 0);
+                var fieldName = fieldDef.Name.IsNil ? null : MetadataReader.GetString(fieldDef.Name);
+                fieldSymbols[index] = new Symbol
+                {
+                    Name = fieldName,
+                    Type = fieldTypeName,
+                    SymbolType = ((fieldDef.Attributes & FieldAttributes.Static) != 0) ? SymbolType.StaticField : SymbolType.Field,
+                    Line = UnknownFieldAndArgLine
+                };
+                index++;
+            }
+
+            return fieldSymbols;
+        }
+
+        private void PopulateMethodScopes(TypeDefinition type, MethodDefinitionHandleCollection methods, Model.Scope[] classMethods, ref int index)
+        {
+            foreach (var methodDefHandle in methods)
+            {
+                if (methodDefHandle.IsNil)
+                {
+                    continue;
+                }
+
+                if (DatadogMetadataReader.IsCompilerGeneratedAttributeDefinedOnMethod(methodDefHandle.RowId))
+                {
+                    continue;
+                }
+
+                var methodDef = MetadataReader.GetMethodDefinition(methodDefHandle);
```

Comment:
Should the error handling policy in potentially be to try/catch and log an error, but ultimately return just skip this item?

Reply:
Check this [comment](https://github.com/DataDog/dd-trace-dotnet/pull/4782#discussion_r1382376151) please

Reply:
Fixed in d76bac0e9bd3cfbda3e480deff9bbd0e453a32ec